### DISCUSSION TITLE
VM listed as "stopped" when started using subprocess.Popen()

### DISCUSSION THREAD

**Discussion by dataprolet** [cirruslabs/tart#968] [General]
I'm trying to write a Python script that starts two VMs for my in the background. To be able to execute the script but keep the VMs running I use the Python module `subprocess.Popen` and redirect stdout and stderr to /dev/null. That leads to `tart list` listing the VMs as stopped and further interaction with them impossible (e. g. `tart ip vm1`). Anybody any idea how I can resolve this?
`subprocess.Popen('/bin/bash -c "/Applications/tart.app/Contents/MacOS/tart run --net-bridged=en0 vm1"', shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)`

**Comment by edigaryev** [comment]
If you want for the child process to survive parent's termination, you need to set `preexec_fn` to either `os.setpgrp` or `os.setsid`, for example:

```python
subprocess.Popen(
    ["tart", "run", "vm"],
    stdin=subprocess.DEVNULL,
    stdout=subprocess.DEVNULL,
    stderr=subprocess.DEVNULL,
    preexec_fn=os.setsid,
)
```
